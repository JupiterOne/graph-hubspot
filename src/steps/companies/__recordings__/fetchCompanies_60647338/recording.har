{
  "log": {
    "_recordingName": "fetchCompanies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "178c54fa7da49daabd661efd712163a1",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            }
          ],
          "headersSize": 387,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "100"
            }
          ],
          "url": "https://api.hubapi.com/crm/v3/objects/companies?limit=100"
        },
        "response": {
          "bodySize": 589,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 589,
            "text": "{\"results\":[{\"id\":\"6534491450\",\"properties\":{\"createdate\":\"2021-07-08T05:09:01.918Z\",\"domain\":\"hubspot.com\",\"hs_lastmodifieddate\":\"2021-07-08T05:10:03.397Z\",\"hs_object_id\":\"6534491450\",\"name\":\"HubSpot, Inc.\"},\"createdAt\":\"2021-07-08T05:09:01.918Z\",\"updatedAt\":\"2021-07-08T05:10:03.397Z\",\"archived\":false},{\"id\":\"6558739580\",\"properties\":{\"createdate\":\"2021-07-13T05:34:55.981Z\",\"domain\":null,\"hs_lastmodifieddate\":\"2021-07-13T05:35:58.142Z\",\"hs_object_id\":\"6558739580\",\"name\":\"Test company\"},\"createdAt\":\"2021-07-13T05:34:55.981Z\",\"updatedAt\":\"2021-07-13T05:35:58.142Z\",\"archived\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 23 Jul 2021 12:41:59 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "6734fc71af87404c-MNL"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "8beafaf1-7d3f-403c-aadb-d7e69c700104"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "97"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "x-trace",
              "value": "2B84998D59240072AA3AF04E74DFF4D086E0CA1DA8000000000000000000"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=QrW8vZPN1Q%2FqopPPvNMzPtN%2FbgHKN85U3ftz2d8Hr98ZiCa%2FnREjUSZmKiidxobFBgMUrKqXeb2BIdrqCcVbGGNxAvxUey2nRCnnt7sprIdP5gguvbUay8qCbXLXmcGd\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1172,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-23T12:42:11.685Z",
        "time": 465,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 465
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
