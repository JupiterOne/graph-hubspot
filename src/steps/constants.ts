import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export enum IntegrationSteps {
  OWNERS = 'fetch-owners',
  COMPANIES = 'fetch-companies',
  DOMAINS = 'fetch-domains',
  TEAMS = 'fetch-teams',
  CONTACTS = 'fetch-contacts',
}

export const Entities: Record<
  'USER' | 'TEAM' | 'CONTACT' | 'COMPANY' | 'DOMAIN',
  StepEntityMetadata
> = {
  /*
     Depending on the other resources later (e.g. will we encounter regular users?)
     We may want to call this "Owner" or similar, we can still use the same class of "User"
     Since user can be both a 'regular' user and the owner.
     A bit hard to tell for sure right now
  */
  USER: {
    _type: 'hubspot_user',
    _class: 'User',
    resourceName: 'HubSpot User',
  },

  TEAM: {
    _type: 'hubspot_team',
    _class: 'Team',
    resourceName: 'HubSpot Team',
  },

  CONTACT: {
    _type: 'hubspot_contact',
    _class: 'User',
    resourceName: 'Hubspot Contact',
  },

  COMPANY: {
    _type: 'hubspot_company',
    _class: 'Organization',
    resourceName: 'HubSpot Company',
  },

  DOMAIN: {
    _type: 'hubspot_domain',
    _class: 'Domain',
    resourceName: 'Hubspot domain',
  },
};

export const Relationships: Record<
  'USER_HAS_TEAM',
  StepRelationshipMetadata
> = {
  USER_HAS_TEAM: {
    _type: 'hubspot_user_has_team',
    _class: RelationshipClass.HAS,
    sourceType: Entities.USER._type,
    targetType: Entities.TEAM._type,
  },
};
