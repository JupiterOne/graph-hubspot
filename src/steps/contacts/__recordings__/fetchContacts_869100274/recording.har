{
  "log": {
    "_recordingName": "fetchContacts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "078af9ceb949dc239ef97569fb49063f",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            }
          ],
          "headersSize": 447,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/crm/v3/objects/contacts"
        },
        "response": {
          "bodySize": 272,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json;charset=utf-8",
            "size": 272,
            "text": "[\"1f8b08000000000000009d904d4bc4301086ffca9293422d493fd635273f2e2278724f2bb24cdba98d244d49522fa5ffdd59a98bb4b0b24286c0cc3bf3ce3c0373e87b1d3c93af035315934cb08875ce76e88242ca0fac7408012b0a2a273c1157fc9ade56a492273213f1669deca80b0d284d92efdf805370dbf485ef6c884b6b48502be7430be630e7f950a75ce3f7b6f8c032ec8fee1a7c30b6\",\"52b5c26aeebad9f25cf21b328ef375ba9bd4d3cc27dbb4deb6ab8b17309dc6d5836d0394e1928dd1cf1177e1f40d7d572d654b537065a33e9136ae417b1ca3095e7e36bd244e33f19b5ed19ca0764fd0da25b5fc0c6c19cf66d81e416bf50effe176dcfe2f6e93eb9cdbdbf805dc7960c282020000\"]"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 08 Jul 2021 05:24:27 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "272"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "66b6e2e6ceb2402d-MNL"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "c38b2aa5-89bf-4cc0-af5b-59ce6540bd39"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "500"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "492"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "50"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "49"
            },
            {
              "name": "x-trace",
              "value": "2BFE9FBAE323C99FB39FAB32D41BAB32279B41DD43000000000000000000"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v2?s=%2BgT1BYwxMPsC%2B50dk2Nx9%2B4FmX8nPUPPccMw1Lc8RYNjnvdjwVBjIvZhLEek3jiMN9U70oCZvZ0vRTFt0k3sJ%2FgQXt1dbvznN5F3B%2BbmCx1AD%2FkIkLUZY39RSw%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1197,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-08T05:24:27.247Z",
        "time": 367,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 367
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
