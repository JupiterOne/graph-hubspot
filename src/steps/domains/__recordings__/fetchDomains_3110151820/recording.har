{
  "log": {
    "_recordingName": "fetchDomains",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "a54c6fc2ec8e026a597e3afad7569300",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            }
          ],
          "headersSize": 454,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/cms/v3/domains"
        },
        "response": {
          "bodySize": 367,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json;charset=utf-8",
            "size": 367,
            "text": "[\"1f8b0800000000000000cd525d4bc33014fd2f79de4a9bba75eb9bca44d0e1d0f9a2f8902dd71a4c93920f758cfd77c3daad212d4e7c1202817b4e6ece3df76c9191867094e30152a02d371ae5cf5bc428cad1284e311ec767139ca0015a2b2006e8b971088e71328c337796499ac7384f47d164943d399aade86f68549684893d27cd26e3641abdd995aea4a948012bcb380515ad65e9a84c2f142b89dadc1241\",\"9928168e8172a32c78d8ccb57363bc12aefdf20597c5426ad3451e9881ba5388dc08f9c9817a9002ca14accd523ac17b45f7a025ff70625a2173222ce17c3327eaddcdef538e3f5c1b53e999202bd7bf7db9afde09bef1988f1ae895543dfa1ba4477f83f4dad460a14d4db93b327c556e60a0978294e0afa950d256eed6ee7f7d585a24c0a0dda0939bf4646ec651164f4fe7e648ebc98d1ed65a7e4a4bb8e3c6872044addb01d09a1d009e7135f21fa212ecbc2bbe2f2861bbc0a16e506ae42f3979d97d03d0c09bbc77040000\"]"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 08 Jul 2021 05:34:53 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "367"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "66b6f22dbe43404b-MNL"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "14c03ec7-6640-433a-819b-f8872c0dab6f"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "500"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "499"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "50"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "49"
            },
            {
              "name": "x-trace",
              "value": "2B309E46AFF20B97AF1AB6390AAF0866EECF4049C2166720D0CDAB1E0501"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v2?s=pAp7t7rhky0h0X6PbFd06Ca6BGLy0LjhFi%2F4t6vfoCIlNrvCMYXQ79AFx68kZsWaGmx%2BnXZSw3BDhWS6jduPtvw7zWA9KJiJK894Xs%2B4yWWlGKigz2lSVHveSg%3D%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1191,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-08T05:34:52.978Z",
        "time": 456,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 456
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
