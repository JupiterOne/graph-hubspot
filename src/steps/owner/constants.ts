export const OwnerEntity = {
  RESOURCE_NAME: 'Owner',
  TYPE: 'acme_user',
  CLASS: 'User',
};
