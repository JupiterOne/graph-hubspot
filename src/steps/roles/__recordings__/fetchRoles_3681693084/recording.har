{
  "log": {
    "_recordingName": "fetchRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "3befe0ca00fe7532a452821224793de3",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "user-agent",
              "value": "hubspot-api-client-nodejs; @VERSION@"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "host",
              "value": "api.hubapi.com"
            },
            {
              "name": "accept",
              "value": "application/json"
            }
          ],
          "headersSize": 376,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.hubapi.com/settings/v3/users/roles"
        },
        "response": {
          "bodySize": 202,
          "content": {
            "mimeType": "application/json;charset=utf-8",
            "size": 202,
            "text": "{\"results\":[{\"id\":\"95124\",\"name\":\"TestRole\",\"requiresBillingWrite\":false},{\"id\":\"95389\",\"name\":\"SuperRole\",\"requiresBillingWrite\":true},{\"id\":\"96735\",\"name\":\"LimitedRole\",\"requiresBillingWrite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Tue, 27 Jul 2021 14:09:59 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "202"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cf-ray",
              "value": "675672d6ee933d94-HKG"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains; preload"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "cf-cache-status",
              "value": "DYNAMIC"
            },
            {
              "name": "access-control-allow-credentials",
              "value": "false"
            },
            {
              "name": "expect-ct",
              "value": "max-age=604800, report-uri=\"https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct\""
            },
            {
              "name": "x-hubspot-correlation-id",
              "value": "9775970e-c743-466b-b6f4-061b04a8e813"
            },
            {
              "name": "x-hubspot-ratelimit-interval-milliseconds",
              "value": "10000"
            },
            {
              "name": "x-hubspot-ratelimit-max",
              "value": "100"
            },
            {
              "name": "x-hubspot-ratelimit-remaining",
              "value": "94"
            },
            {
              "name": "x-hubspot-ratelimit-secondly",
              "value": "10"
            },
            {
              "name": "x-hubspot-ratelimit-secondly-remaining",
              "value": "9"
            },
            {
              "name": "report-to",
              "value": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=D9B%2BY63zoytJh4NRFBx%2FVolE3gdbprrNL1LmzFx%2Fd960l%2BT7VJNXiCmk25MbdTer2AZt%2FAioFhutmY9zP7PbMASgSBtshGoM%2F3T2GuusQ7gRSx2c1ZINoRLF4z6V1HmM\"}],\"group\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "nel",
              "value": "{\"report_to\":\"cf-nel\",\"max_age\":604800}"
            },
            {
              "name": "server",
              "value": "cloudflare"
            },
            {
              "name": "alt-svc",
              "value": "h3-27=\":443\"; ma=86400, h3-28=\":443\"; ma=86400, h3-29=\":443\"; ma=86400, h3=\":443\"; ma=86400"
            }
          ],
          "headersSize": 1100,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-27T14:10:07.160Z",
        "time": 506,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 506
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
