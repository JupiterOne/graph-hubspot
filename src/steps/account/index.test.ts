import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import * as account from '.';
import { createIntegrationConfig } from '../../../test/config';
import { IntegrationConfig } from '../../config';
import { Entities } from '../constants';
import { setupHubspotRecording } from '../../../test/recording';

describe('#fetchAccount', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupHubspotRecording({
      directory: __dirname,
      name: 'fetchAccount',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: createIntegrationConfig(),
    });

    await account.fetchAccount(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const companies = context.jobState.collectedEntities.filter((e) =>
      e._class.includes(Entities.ACCOUNT._class as string),
    );

    expect(companies.length).toBeGreaterThan(0);
    expect(companies).toMatchGraphObjectSchema({
      _class: [Entities.ACCOUNT._class as string],
      schema: {
        additionalProperties: false,
        required: [],
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: Entities.ACCOUNT._type },

          name: { type: 'string' },
          accessUrl: { type: 'string' },
        },
      },
    });
  });
});
