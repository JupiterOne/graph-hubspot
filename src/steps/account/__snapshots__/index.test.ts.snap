// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchAccount should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "hubspot_account:334798 ",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiBaseUrl": "https://api.hubapi.com",
            "appId": "334798",
          },
        },
      ],
      "_type": "hubspot_account",
      "accessUrl": "https://api.hubapi.com",
      "createdOn": undefined,
      "displayName": "334798",
      "name": "334798",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "hubspot_account",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
